demos/object_detection_demo/cpp/main.cpp

```diff
@@ -402,32 +401,35 @@ int main(int argc, char *argv[]) {
                     int key = cv::waitKey(1);
                     if (27 == key || 'q' == key || 'Q' == key) {  // Esc
                         keepRunning = false;
-                    }
-                    else {
+                    } else {
                         presenter.handleKey(key);
                     }
                 }
             }
-        }
+        } // while(keepRunning)
 
         //// ------------ Waiting for completion of data processing and rendering the rest of results ---------
         pipeline.waitForTotalCompletion();
+
         for (; framesProcessed <= frameNum; framesProcessed++) {
-            while (!(result = pipeline.getResult())) {}
-            auto renderingStart = std::chrono::steady_clock::now();
-            cv::Mat outFrame = renderDetectionData(result->asRef<DetectionResult>(), palette, outputTransform);
-            //--- Showing results and device information
-            presenter.drawGraphs(outFrame);
-            renderMetrics.update(renderingStart);
-            metrics.update(result->metaData->asRef<ImageMetaData>().timeStamp,
-                outFrame, { 10, 22 }, cv::FONT_HERSHEY_COMPLEX, 0.65);
-            if (videoWriter.isOpened() && (FLAGS_limit == 0 || framesProcessed <= FLAGS_limit - 1)) {
-                videoWriter.write(outFrame);
-            }
-            if (!FLAGS_no_show) {
-                cv::imshow("Detection Results", outFrame);
-                //--- Updating output window
-                cv::waitKey(1);
+            result = pipeline.getResult();
+            if (result != nullptr)
+            {
+                auto renderingStart = std::chrono::steady_clock::now();
+                cv::Mat outFrame = renderDetectionData(result->asRef<DetectionResult>(), palette, outputTransform);
+                //--- Showing results and device information
+                presenter.drawGraphs(outFrame);
+                renderMetrics.update(renderingStart);
+                metrics.update(result->metaData->asRef<ImageMetaData>().timeStamp,
+                    outFrame, { 10, 22 }, cv::FONT_HERSHEY_COMPLEX, 0.65);
+                if (videoWriter.isOpened() && (FLAGS_limit == 0 || framesProcessed <= FLAGS_limit - 1)) {
+                    videoWriter.write(outFrame);
+                }
+                if (!FLAGS_no_show) {
+                    cv::imshow("Detection Results", outFrame);
+                    //--- Updating output window
+                    cv::waitKey(1);
+                }
```

Comment:
This will not work in case of slow networks. getResult would return nullptr if result is not ready yet and body of "if" will not be executed, but framesProcessed will be increased - as in case if we got processed data. So, we'll miss results from some submissions as framesProcessed would reach frameNum before all submissions are processed.
Please revert this part as it was before.

Reply:
that what we have in segmentation_demo

Reply:
yes, you're right - after waitForTotalCOmpletion  this code will work. So we can leave it as is.